test/functional/Valhalla/src/org/openj9/test/lworld/ValueTypeTests.java

```diff
@@ -93,6 +93,63 @@ static public void testCreatePoint2D() throws Throwable {
 		assertEquals(getY.invoke(point2D), yNew);
 	}
 
+	/*
+	 * Test array of Point2D
+	 */
+	@Test(priority=1)
+	static public void testCreateArrayPoint2D() throws Throwable {
+		String fields[] = {"x:I", "y:I"};
+		point2DClass = ValueTypeGenerator.generateValueClass("Point2D", fields);
+		
+		makePoint2D = lookup.findStatic(point2DClass, "makeValue", MethodType.methodType(point2DClass, int.class, int.class));
+		
+		getX = generateGetter(point2DClass, "x", int.class);
+		MethodHandle withX = generateWither(point2DClass, "x", int.class);
+		getY = generateGetter(point2DClass, "y", int.class);
+		MethodHandle withY = generateWither(point2DClass, "y", int.class);
+
+		int x1 = 0xFFEEFFEE;
+		int y1 = 0xAABBAABB;
+		int xNew1 = 0x11223344;
+		int yNew1 = 0x99887766;
+		
+		Object point2D_1 = makePoint2D.invoke(x1, y1);
+		
+		assertEquals(getX.invoke(point2D_1), x1);
+		assertEquals(getY.invoke(point2D_1), y1);
+		
+		point2D1 = withX.invoke(point2D_1, xNew1);
+		point2D1 = withY.invoke(point2D_1, yNew1);
+		
+		assertEquals(getX.invoke(point2D_1), xNew1);
+		assertEquals(getY.invoke(point2D_1), yNew1);
+
+
+		int x2 = 0x00000011;
+		int y2 = 0xABCDEF00;
+		int xNew2 = 0x11111111;
+		int yNew2 = 0xFFFFFFFF;
+		
+		Object point2D_2 = makePoint2D.invoke(x2, y2);
+		
+		assertEquals(getX.invoke(point2D_2), x2);
+		assertEquals(getY.invoke(point2D_2), y2);
+		
+		point2D1 = withX.invoke(point2D_2, xNew2);
+		point2D1 = withY.invoke(point2D_2, yNew2);
+		
+		assertEquals(getX.invoke(point2D_2), xNew2);
+		assertEquals(getY.invoke(point2D_2), yNew2);
+
+		Class arrayClass = Array.newInstance(point2DClass, 3).getClass;
+		Array.set(arrayClass, 1, point2D_1);
+		Array.set(arrayClass, 2, point2D_2);
+
+		assertEquals(Array.get(arrayClass, 1), point2D_1);
+		assertEquals(Array.get(arrayClass, 2), point2D_2);
+		assertEquals(Array.get(arrayClass, 3), null);
+	}
+
```

Comment:
We will also need a test for FlattenedLine2D that does something similar to what you have above